runtime/vm/createramclass.cpp

```diff
@@ -2549,6 +2565,114 @@ initializeClassLinks(J9Class *ramClass, J9Class *superclass, J9MemorySegment *se
+
+	if (NULL != nlsMessage) {
+#define TYPE_CLASS "class"
+#define TYPE_INTERFACE "interface"
+#define SUPER_CLASS "superclass"
+#define SUPER_INTERFACE "superinterface"
+#if JAVA_SPEC_VERSION == 8
+		UDATA errorMsgLen = j9str_printf(NULL, 0, nlsMessage,
+				isSubClassInterface ? TYPE_INTERFACE : TYPE_CLASS,
+				J9UTF8_LENGTH(subClassName), J9UTF8_DATA(subClassName),
+				J9UTF8_LENGTH(subClassLoaderName), J9UTF8_DATA(subClassLoaderName),
+				isSuperclass ? SUPER_CLASS : SUPER_INTERFACE,
+				J9UTF8_LENGTH(superClassOrInterfaceName), J9UTF8_DATA(superClassOrInterfaceName),
+				J9UTF8_LENGTH(superClassOrInterfaceLoaderName), J9UTF8_DATA(superClassOrInterfaceLoaderName));
+#else /* JAVA_SPEC_VERSION == 8 */
+		UDATA errorMsgLen = j9str_printf(NULL, 0, nlsMessage,
+				isSubClassInterface ? TYPE_INTERFACE : TYPE_CLASS,
+				J9UTF8_LENGTH(subClassName), J9UTF8_DATA(subClassName),
+				subClassModuleName,
+				J9UTF8_LENGTH(subClassLoaderName), J9UTF8_DATA(subClassLoaderName),
+				isSuperclass ? SUPER_CLASS : SUPER_INTERFACE,
+				J9UTF8_LENGTH(superClassOrInterfaceName), J9UTF8_DATA(superClassOrInterfaceName),
+				superClassOrInterfaceModuleName,
+				J9UTF8_LENGTH(superClassOrInterfaceLoaderName), J9UTF8_DATA(superClassOrInterfaceLoaderName));
+#endif /* JAVA_SPEC_VERSION == 8 */
+		errorMsg = (char*)j9mem_allocate_memory(errorMsgLen, OMRMEM_CATEGORY_VM);
+		if (NULL == errorMsg) {
+			j9object_t detailMessage = currentThread->javaVM->memoryManagerFunctions->j9gc_createJavaLangString(currentThread, J9UTF8_DATA(subClassName), J9UTF8_LENGTH(subClassName), J9_STR_XLAT);
+			setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALACCESSERROR, (UDATA *)detailMessage);
+			return;
+		}
+#if JAVA_SPEC_VERSION == 8
+		j9str_printf(errorMsg, errorMsgLen, nlsMessage,
+				isSubClassInterface ? TYPE_INTERFACE : TYPE_CLASS,
+				J9UTF8_LENGTH(subClassName), J9UTF8_DATA(subClassName),
+				J9UTF8_LENGTH(subClassLoaderName), J9UTF8_DATA(subClassLoaderName),
+				isSuperclass ? SUPER_CLASS : SUPER_INTERFACE,
+				J9UTF8_LENGTH(superClassOrInterfaceName), J9UTF8_DATA(superClassOrInterfaceName),
+				J9UTF8_LENGTH(superClassOrInterfaceLoaderName), J9UTF8_DATA(superClassOrInterfaceLoaderName));
+#else /* JAVA_SPEC_VERSION == 8 */
+		j9str_printf(errorMsg, errorMsgLen, nlsMessage,
+				isSubClassInterface ? TYPE_INTERFACE : TYPE_CLASS,
+				J9UTF8_LENGTH(subClassName), J9UTF8_DATA(subClassName),
+				subClassModuleName,
+				J9UTF8_LENGTH(subClassLoaderName), J9UTF8_DATA(subClassLoaderName),
+				isSuperclass ? SUPER_CLASS : SUPER_INTERFACE,
+				J9UTF8_LENGTH(superClassOrInterfaceName), J9UTF8_DATA(superClassOrInterfaceName),
+				superClassOrInterfaceModuleName,
+				J9UTF8_LENGTH(superClassOrInterfaceLoaderName), J9UTF8_DATA(superClassOrInterfaceLoaderName));
+#endif /* JAVA_SPEC_VERSION == 8 */
+#undef TYPE_CLASS
+#undef TYPE_INTERFACE
+#undef SUPER_INTERFACE
+#undef SUPER_CLASS
+	}
+
+#if JAVA_SPEC_VERSION >= 11
```

Comment:
This condition should also be `JAVA_SPEC_VERSION != 8`.

Reply:
Updated